ml-agents/mlagents/trainers/ppo/models.py

```diff
@@ -33,15 +31,20 @@ def __init__(
         :param h_size: Size of hidden layers
         :param epsilon: Value for policy-divergence threshold.
         :param beta: Strength of entropy regularization.
-        :return: a sub-class of PPOAgent tailored to the environment.
         :param max_step: Total number of training steps.
         :param normalize: Whether to normalize vector observation input.
         :param use_recurrent: Whether to use an LSTM layer in the network.
         :param num_layers Number of hidden layers between encoded input and policy & value layers
         :param m_size: Size of brain memory.
+        :param seed: Seed to use for initialization of model. 
+        :param stream_names: List of names of value streams. Usually, a list of the Reward Signals being used. 
+        :return: a sub-class of PPOAgent tailored to the environment.
         """
-        LearningModel.__init__(self, m_size, normalize, use_recurrent, brain, seed)
-        self.use_curiosity = use_curiosity
+        if stream_names is None:
```

Comment:
Is it ever really valid to have no stream names? Will the code run without any value heads?

Reply:
Yes, in the BC Trainer (not BC Module) the Trainer base class is instantiated with no reward signals. Only the Policy part of the network is used. 

However, in PPO this isn't valid, and in fact this code is duplicated in the parent Trainer. I've removed this check, and added one in the PPO policy that throws an exception if no reward signals are defined.